packages/ui/src/components/language-selector/index.tsx

```diff
@@ -0,0 +1,130 @@
+}
+
+export interface LanguageInterface {
+  code: LanguageCode
+  name: Language
+}
+
+const languages: LanguageInterface[] = [
+  { code: LanguageCode.EN, name: Language.English },
+  { code: LanguageCode.БГ, name: Language.Bulgarian },
+  { code: LanguageCode.HR, name: Language.Croatian },
+  { code: LanguageCode.CZ, name: Language.Czech },
+  { code: LanguageCode.FR, name: Language.French },
+  { code: LanguageCode.DE, name: Language.German },
+  { code: LanguageCode.IE, name: Language.Irish },
+  { code: LanguageCode.PY, name: Language.Russian },
+  { code: LanguageCode.LA, name: Language.LatinAmerican }
+]
+
+interface LanguageDialogProps {
+  defaultLanguage?: LanguageCode
+  language?: LanguageCode
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onChange: (language: LanguageInterface) => void
+  onSave: (language: LanguageInterface) => void
+  onCancel: () => void
+  children: React.ReactNode
+}
+
+const LanguageDialog: React.FC<LanguageDialogProps> = ({
+  defaultLanguage,
+  language,
+  open,
+  onOpenChange,
+  onChange,
+  onSave,
+  onCancel,
+  children
+}) => {
+  const [selectedLanguage, setSelectedLanguage] = useState<LanguageCode | null>(null)
+
+  useEffect(() => {
+    if (language) {
+      setSelectedLanguage(language)
+    } else if (defaultLanguage) {
+      setSelectedLanguage(defaultLanguage)
+    }
+  }, [defaultLanguage, language])
+
+  const handleSave = (): void => {
+    if (selectedLanguage) {
+      const languageToSave = languages.find(lang => lang.code === selectedLanguage)
+      if (languageToSave) {
+        onSave(languageToSave)
+      }
+    }
+  }
+
+  return (
+    <Dialog.Root open={open} onOpenChange={onOpenChange}>
+      <Dialog.Trigger asChild>{children}</Dialog.Trigger>
+      <Dialog.Content className="w-[400px]">
+        <Dialog.Title>Language</Dialog.Title>
+        <div className="flex flex-col gap-3">
+          {languages.map(lang => (
+            <Button
+              variant="ghost"
+              key={lang.code}
+              className="flex justify-between items-center py-2 px-1 rounded-md cursor-pointer hover:bg-gray-400"
+              onClick={() => {
+                setSelectedLanguage(lang.code)
+                onChange(lang)
+              }}
+            >
+              <div className="flex items-center gap-2">
+                <div className="w-7 h-7 flex justify-center items-center bg-background-12 rounded-sm">{lang.code}</div>
+                <span>{lang.name}</span>
+              </div>
+              {selectedLanguage === lang.code && <CheckIcon width={20} height={20} />}
+            </Button>
+          ))}
+        </div>
+
+        {/* Buttons */}
+        <Dialog.Footer>
+          <ButtonGroup>
+            <>
+              <Button variant="ghost" onClick={onCancel}>
+                Cancel
+              </Button>
+              <Button variant="default" onClick={handleSave}>
+                Save
+              </Button>
+            </>
```

Comment:
No need for the fragment
```suggestion
            <Button variant="ghost" onClick={onCancel}>
              Cancel
            </Button>
            <Button variant="default" onClick={handleSave}>
              Save
            </Button>
```

Reply:
Done, removed it.